### ISSUE TITLE
Uncaught TypeError: validateEventDispatches is not a function

### ISSUE THREAD

**Issue by nmn** [facebook/nuclide#565]
Hyperclick:

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.8.0
**System**: Mac OS X 10.11.4
**Thrown From**: [hyperclick](https://github.com/facebooknuclideapm/hyperclick) package, v0.0.36

### Stack Trace

Uncaught TypeError: validateEventDispatches is not a function

```
At /Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/EventPluginUtils.js:99

TypeError: validateEventDispatches is not a function
  at Object.executeDispatchesInOrder (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/EventPluginUtils.js:99:5)
  at executeDispatchesAndRelease (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/EventPluginHub.js:42:22)
  at executeDispatchesAndReleaseTopLevel (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/EventPluginHub.js:53:10)
  at Array.forEach (native)
  at forEachAccumulated (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/forEachAccumulated.js:24:9)
  at Object.EventPluginHub.processEventQueue (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/EventPluginHub.js:215:7)
  at runEventQueueInBatch (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/ReactEventEmitterMixin.js:18:18)
  at Object.ReactEventEmitterMixin.handleTopLevel [as _handleTopLevel] (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/ReactEventEmitterMixin.js:29:5)
  at handleTopLevelImpl (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/ReactEventListener.js:73:24)
  at ReactDefaultBatchingStrategyTransaction.Mixin.perform (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/Transaction.js:136:20)
  at Object.ReactDefaultBatchingStrategy.batchedUpdates (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/ReactDefaultBatchingStrategy.js:63:19)
  at Object.batchedUpdates (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/ReactUpdates.js:97:20)
  at ReactEventListener.dispatchEvent (/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/node_modules/react/lib/ReactEventListener.js:150:20)

```

### Commands

```
     -0:24.9.0 core:save (atom-text-editor.editor.is-focused)
     -0:15.9.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-when-scrolling.use-animations.theme-atom-material-syntax.theme-atom-material-ui)
```

### Config

```json
{
  "core": {
    "closeEmptyWindows": false,
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "destroyEmptyPanes": false,
    "disabledPackages": [
      "js-autoimport",
      "atom-linter-node-dependencies",
      "autocomplete-java",
      "tree-view",
      "pull-requests"
    ],
    "excludeVcsIgnoredPaths": false,
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
Stylus, v3.0.0
angularjs, v0.3.4
atom-codemod, v1.2.0
atom-htmltidy, v5.1.0
atom-material-syntax, v0.4.6
atom-material-ui, v1.2.13
atom-reason-loader, v0.1.0
autocomplete-flow, v1.5.0
autocomplete-json, v5.0.3
autocomplete-python, v1.8.1
debug, v0.0.1
emmet, v2.4.3
haskell-grammar, v0.4.0
highlight-selected, v0.11.2
java-importer, v1.4.1
js-hyperclick, v1.4.2
language-elm, v1.5.0
language-gherkin, v1.0.4
language-graphql, v0.2.0
language-ini, v1.16.0
language-ocaml, v1.1.2
language-reason, v0.0.9
language-thrift, v1.0.2
linter, v1.11.14
linter-eslint, v7.2.4
linter-flow, v5.4.0
linter-javac, v1.9.4
linter-tslint, v0.10.0
merge-conflicts, v1.4.4
minimap, v4.24.6
minimap-git-diff, v4.3.1
minimap-split-diff, v0.3.1
node-debugger, v1.7.0
pretty-json, v1.4.1
standard-formatter, v2.0.1
tool-bar, v0.4.0

# Dev
No dev packages
```

**Comment by hansonw** [comment]
Seems like some kind of React version incompatibility - @ssorallen or @zertosh, have you seen this kind of issue before?

**Comment by zertosh** [comment]
Looks like two packages may be trying to use different versions of React. @nmn can you run `grep -r --include='package.json' '"react-for-atom"' ~/.atom/packages` to see which packages it is?

**Comment by nmn** [comment]
@zertosh you seem to right!

```
/Users/ngoel/.atom/packages/hyperclick/node_modules/react-for-atom/package.json:  "name": "react-for-atom",
/Users/ngoel/.atom/packages/hyperclick/package.json:    "react-for-atom": "15.0.1-0"
/Users/ngoel/.atom/packages/hyperclick/package.json:        "name": "react-for-atom",
/Users/ngoel/.atom/packages/hyperclick/package.json:          "react-for-atom": "15.0.1-0"
/Users/ngoel/.atom/packages/nuclide/node_modules/react-for-atom/package.json:  "name": "react-for-atom",
/Users/ngoel/.atom/packages/nuclide/package.json:    "react-for-atom": "15.0.2-1",
/Users/ngoel/.atom/packages/nuclide/package.json:        "name": "react-for-atom",
/Users/ngoel/.atom/packages/nuclide/package.json:          "react-for-atom": "15.0.2-1",
```

**Comment by zertosh** [comment]
We found the problem. The version of react-for-atom that hyperclick@0.0.36 was using had this issue https://github.com/jgebhardt/react-for-atom/issues/13. I just published hyperclick@0.0.37 with the fixed react-for-atom. If you updated hyperclick and reload Atom, you should be fine.

**Comment by ssorallen** [comment]
Hyperclick v0.0.37 resolves this issue.

Note that Nuclide bundles Hyperclick. You shouldn't need to have both installed, and having both might lead to strange behavior. It shouldn't, however, lead to exceptions like you were seeing, and that has been fixed in v0.0.37.
